sklearn/tests/test_discriminant_analysis.py

```diff
@@ -176,8 +175,12 @@ def test_lda_explained_variance_ratio():
     clf_lda_svd = LinearDiscriminantAnalysis(solver="svd")
     clf_lda_svd.fit(X, y)
     assert_almost_equal(clf_lda_svd.explained_variance_ratio_.sum(), 1.0, 3)
+
+    # NOTE: clf_lda_eigen.explained_variance_ratio_ is not of n_components
+    # length. Make it the same length as clf_lda_svd.explained_variance_ratio_
+    # before comparison.
     assert_array_almost_equal(clf_lda_svd.explained_variance_ratio_,
-                              clf_lda_eigen.explained_variance_ratio_)
+            clf_lda_eigen.explained_variance_ratio_[:3])
```

Comment:
I mean, why is the sdv array of length 3 ? We don't specify n_components when we initialize the classifiers.